### DISCUSSION TITLE
Record-syntax issues

### DISCUSSION THREAD

**Discussion by TobiasWolters** [saucecontrol/InheritDoc#8] [Q&A] [answered]
Consider the following case, using the new record-syntax from C# 9.0:
```csharp
public class A
{
    /// <summary>
    ///     Int comment
    /// </summary>
    public int Int { get; set; }
}

/// <summary>
///     B documentation
/// </summary>
/// <param name="Int"><inheritdoc cref="A.Int"/></param>
public record B(int Int);
```
With the current version (1.3.0) of `InheritDoc`, this results in the following warnings and does not follow the `inheritdoc`:

- warning CS1574: XML comment has cref attribute 'Int' that could not be resolved
- InheritDocTask warning IDT002: No matching documentation could be found for: T:InheritDocTest.B, which attempts to inherit from: !:A.Int
- No matching documentation could be found for: M:InheritDocTest.B.#ctor(System.Int32), which attempts to inherit from: !:A.Int

Ideally, `InheritDoc` should be able to resolve the inheritdoc to '`Int comment`' without any warnings.

**Comment by saucecontrol** [comment, marked as answer]
Looks like you have a couple of problems there...

The first warning you've got (`CS1574`) is from the C# compiler, saying it can't resolve the `cref`.  When that happens, the identifier in the comment is prefixed with `!:` (as defined in the [spec](https://github.com/dotnet/csharplang/blob/main/spec/documentation-comments.md#id-string-format)), which then means InheritDocTask can't resolve the reference.  I am unable to reproduce that with the .NET6 RC1 SDK, so that would appear to be a Roslyn bug that has been resolved.

The second issue you have is that you're trying to inherit a `summary` into a `param`.  You must give an explicit path in this case so that you reach into the *content* of the summary, which would make that doc line `<param name="Int"><inheritdoc cref="A.Int" path="/summary/node()"/></param>`.

Changing the above and using the latest C# compiler results in no errors and the following output:

```XML
<?xml version="1.0" encoding="utf-8"?>
<doc>
    <members>
        <member name="P:A.Int">
            <summary>
                Int comment
            </summary>
        </member>
        <member name="T:B">
            <summary>
                B documentation
            </summary>
            <param name="Int">
                Int comment
            </param>
        </member>
        <member name="M:B.#ctor(System.Int32)">
            <summary>
                B documentation
            </summary>
            <param name="Int">
                Int comment
            </param>
        </member>
    </members>
</doc>
```

**Reply by TobiasWolters** [reply]
Thanks a lot for taking your time to answer this! 
That makes perfectly sense 👍
